### PR TITLE
Remove random things from versioning.md

### LINKED ISSUE #17807: Remove API versioning info from versioning.md
@bgrant0607, can you please add the information you think is necessary to [versioning.md](https://github.com/kubernetes/kubernetes/blob/master/docs/design/versioning.md)?  There's some info about how API versioning relates to releases, but it should probably be expanded?  (Or perhaps this information already exists elsewhere, and should just be linked in.)  (Came up while I was writing #17806.)

### PR DESCRIPTION
**Pull request by lavalamp**
Fixes #17807

cc @bgrant0607 for fact checking, if desired

### CODE REVIEW

File: docs/design/versioning.md [hunk 1]
```diff
@@ -57,19 +57,31 @@ There is no mandated timeline for major versions. They only occur when we need t
...
-### Rationale for API v2 being complete before v2.0's release
-
-It may seem a bit strange to complete the v2 API before v2.0 is released, but *adding* a v2 API is not a breaking change. *Removing* the v2beta\* APIs *is* a breaking change, which is what necessitates the major version bump. There are other ways to do this, but having the major release be the fresh start of that release's API without the baggage of its beta versions seems most intuitive out of the available options.
+Release versions and API versions are entirely separate. API versions apply only
+to a particular API group. We can say a few things, though:
+
+* API versions will not go *backwards*!
+* API versions have a defined lifecycle, going through alpha and beta
+  incarnations before becoming stable.
+  * group/v1alpha1 -> group/v1beta1 -> group/v1 -> group/v2alpha1 ...
```
Comment (ihmccreery) [inline docs/design/versioning.md]: Presumably some of these will be available at the same time, but not others?  E.g. `v1beta1` replaces `v1alpha1`, but `v2alpha1` is available in addition to `v1`?

Comment (ihmccreery) [inline docs/design/versioning.md]: Ah, I see you touch on this later.  But this bullet is a bit confusing nonetheless.

File: docs/design/versioning.md [hunk 2]
```diff
@@ -57,19 +57,31 @@ There is no mandated timeline for major versions. They only occur when we need t
...
+  will be adequate time to update clients.**
+* Clients will need to understand only one version per group. E.g., net/v2 will
+  completely replace net/v1. Both will be served simultaneously by some version
+  of kubernetes--this is to allow for a transition period with both old and
+  new clients.
+* (For clarity: The default API group, for historical reasons, is "" (empty
+  string). This will become less important as we move more things into their own
+  groups.)
+
+For more, see the [api documentation](../api.md#api-versioning).
```
Comment (ihmccreery) [inline docs/design/versioning.md]: Ah ha!  I'd just move as much of this as is reasonable into api.md#api-versioning, (I didn't realize that existed,) and just link to it from here.

**Comment by ihmccreery** [comment]
LGTM with comments.  Feel free to apply label once they're addressed.

**Comment by fejta** [comment]
This PR has had no meaningful activity for multiple months. If it is still valid please rebase, push a new commit and reopen the PR. Thanks!

**Pull request closed without merging**